network/peer/peer.go

```diff
@@ -819,7 +826,17 @@ func (p *peer) getUptime() uint32 {
 	return primaryUptimePercent
 }
 
-func (*peer) handlePong(*p2p.Pong) {}
+func (p *peer) handlePong(*p2p.Pong) {
+	lastPing := atomic.LoadInt64(&p.lastPingSent)
+	if lastPing == 0 {
+		return
+	}
+
+	elapsed := p.Clock.Time().UnixMilli() - lastPing
```

Comment:
Seems like we might as well enforce that we actually sent a ping.
```suggestion
	pingSent := atomic.SwapInt64(&p.lastPingSent, 0)
	if pingSent == 0 {
		p.Log.Debug(malformedMessageLog,
			zap.Stringer("nodeID", p.id),
			zap.Stringer("messageOp", message.PongOp),
			zap.String("reason", "received unexpected pong"),
		)
		p.StartClose()
		return
	}

	elapsed := p.Clock.Time().UnixMilli() - pingSent
```